include/rotatingMHD/magnetic_induction.h

```diff
@@ -0,0 +1,490 @@
+TensorFunction<1,dim>                                 *ptr_velocity_function;
+
+/*!
+ * @brief A method for the set-up of the solver's linear algebra.
+ *
+ */
+void setup() override;
+
+/*!
+ * @brief A method for the set-up of the pseudo-pressure field.
+ *
+ */
+void setup_pseudo_pressure();
+
+/*!
+ * @brief A method for the set-up of auxiliary scalar field.
+ *
+ */
+void setup_auxiliary_scalar() override;
+
+/*!
+ * @brief A method, which sets up the matrices of the solver
+ *
+ */
+void setup_matrices() override;
+
+/*!
+ * @brief A virtual method, which sets up the vectors of the solver
+ *
+ */
+void setup_vectors() override;
+
+/*!
+ * @brief A method that computes admissable initial
+ * conditions for the pseudo-pressure by considering and solving
+ * the system of equations in its steady-state at \f$ t = 0 \f$.
+ *
+ * @details The initialization step consist of the numerical solution
+ * of the continuous variational problem: Find \f$\sigma^0 \in V_\sigma \f$
+ * such that
+ * \f[
+ * \hat{a}_\textrm{init}(q, \sigma^k) = \hat{L}_\textrm{init}(q),
+ * \qquad \forall q \in V_{q},
+ * \f]
+ * with
+ * \f[
+ * \hat{a}_\textrm{init}(q, \sigma^0)
+ * =
+ * \int_\Omega \nabla q \cdot \nabla \sigma^0 \dint{v}
+ * \quad \textrm{and}\quad
+ * \hat{L}_\textrm{init}(q) =- \int_{\partial \Omega}
+ * \frac{1}{\magPrandtl} q \nabla^2 \bs{B}^0 \cdot \bs{n }\dint{a} -
+ * \int_\Omega \nabla q \cdot \bs{f}^0\dint{v};
+ * \f]
+ *
+ */
+void initialization_step() override;
+
+/*!
+ * @brief A method performing the diffusion step.
+ *
+ * @param reinit_preconditioner A boolean indicating if the
+ * preconditioner is to be re-built.
+ *
+ * @details The diffusion step consist of the numerical solution of the
+ * continuous variational problem: Find \f$\boldsymbol{\mathfrak{B}}^k
+ * \in V_{\boldsymbol{\mathfrak{B}}}\f$ such that
+ * \f[
+ * \hat{a}_\textrm{diff}(\bs{q}, \boldsymbol{\mathfrak{B}}^k) =
+ * \hat{L}_\textrm{diff}(\bs{q}), \qquad \forall \bs{q} \in V_{\bs{q}},
+ * \f]
+ * with
+ * \f[
+ * \hat{a}_\textrm{diff}(\bs{q}, \boldsymbol{\mathfrak{B}}^k)
+    =
+    \int_\Omega \left[ \frac{\alpha_0^k}{\Delta t^k} \bs{q} \cdot
+    \boldsymbol{\mathfrak{B}}^k + \frac{\gamma_0^k}{\magPrandtl}
+    (\nabla \times \bs{q}) \cdot (\nabla \times \boldsymbol{\mathfrak{B}}^k  )
+    \right] \dint{v}
+ * \f]
+ * and
+ * \f{align}{
+ * \hat{L}_\textrm{diff}(\bs{q}) =& \int_\Omega \bs{q}
+ * \cdot \nabla \sigma^{\sharp, k}\dint{v}\\& -
+ * \int_\Omega \sum_{j=1}^s \frac{\alpha_j^{k}}{\Delta t^k} \bs{q}
+ * \cdot \boldsymbol{\mathfrak{B}}^{k-j}  \dint{v} \\
+ *   & + \int_\Omega \sum_{j=1}^s \beta_j^k \bs{q} \cdot [ (\nabla
+ *    \cdot \boldsymbol{\mathfrak{B}}^{k-j}) \bs{v}^{k-j} +
+ *    (\nabla \cdot \bs{v}^{k-j}) \boldsymbol{\mathfrak{B}}^{k-j}
+ *    \\ & \qquad \qquad \qquad \quad + \boldsymbol{\mathfrak{B}}^{k-j}
+ *    \cdot (\nabla \otimes \bs{v}^{k-j}) -
+ *    \bs{v}^{k-j} \cdot (\nabla \otimes \boldsymbol{\mathfrak{B}}^{k-j})]
+ *    \dint{v} \\
```

Comment:
The div(B) and div(v) should be removed because of the incompressibility constraint.